go/vt/vtgate/gateway/discoverygateway.go

```diff
@@ -84,7 +84,11 @@ type discoveryGateway struct {
 	buffer *buffer.Buffer
 }
 
-func createDiscoveryGateway(hc discovery.HealthCheck, topoServer *topo.Server, serv srvtopo.Server, cell string, retryCount int) Gateway {
+func createDiscoveryGateway(hc discovery.HealthCheck, serv srvtopo.Server, cell string, retryCount int) Gateway {
+	var topoServer *topo.Server
+	if serv != nil {
+		topoServer = serv.GetTopoServer()
+	}
```

Comment:
Looking at this file, you can also just remove the topoServer member variable in discoveryGateway, as it's unused. That way, when anyone wants to use the topoServer, they'll just use srvtopo.Server.GetTopoServer() instead.

Reply:
Oh good point. Done!